pace-util/pace/util/zarr_monitor.py

```diff
@@ -126,6 +126,16 @@ def store(self, state: dict) -> None:
         for name, quantity in sorted(state.items(), key=lambda x: x[0]):
             self._writers[name].append(quantity)  # type: ignore[index]
 
+    def store_constant(self, grid: dict) -> None:
+        for name, quantity in grid.items():
```

Comment:
Might need some imports to make this code suggestion work. You also might want to test what happens when you do this, I'm not sure if it will error (due to the store already being created) or overwrite it. If it errors you could consider adding some kind of running tab of what constants have been stored, and raising a more helpful exception instead of adding this docstring note. My note isn't great because if there's a docstring about the errors they aren't exactly "unexpected".

```suggestion
    """
    Store a state dictionary containing values which do not vary in time.
    
    This function can be called at most once for a given state key,
    if you try to store the same constant multiple times you may
    get unexpected errors.
    """
    def store_constant(self, state: Mapping[str, Quantity]) -> None:
        for name, quantity in state.items():
```